libqtile/widget/systray.py

```diff
@@ -107,16 +109,16 @@ class Systray(window._Window, base._Widget):
     def __init__(self, **config):
         base._Widget.__init__(self, bar.CALCULATED, **config)
         self.add_defaults(Systray.defaults)
-        self.icons = {}
+        self.tray_icons = []
```

Comment:
A list doesn't feel right here because it isn't fully type safe. If a misbehaving client sends 2 ClientMessages the list will end up with 2 icons for the same `wid`, which will then cause problems. I'm thinking that that was actually a benefit to using the wids as dictionary keys. So perhaps it could be `Dict[int, Icon]` with wid keys, and then we can sort it with:
```py
self.tray_icons = dict(sorted(self.tray_icons.items(), key=lambda item: item[1].name))
```
It's kinda ugly... but it's a bit more safe I think.

Reply:
If you're assuming that multiple messages have the same `wid` then the Icon object should be the same so we could add an `__eq__` method to `Icon` and then just do:
``` python
if icon not in self.tray_icons:
```
That would look a bit neater than your line (but is more code)!

Reply:
Interesting. Yeah that sounds neat.